tests/Unit/PointwiseFunctions/Xcts/Test_AdmMomentum.cpp

```diff
@@ -249,37 +280,56 @@ void test_infinite_volume_integral(const double distance, const double mass,
                                       inv_spatial_metric(ti::J, ti::L) *
                                       extrinsic_curvature(ti::k, ti::l));
 
-    const auto surface_integrand = Xcts::adm_linear_momentum_surface_integrand(
-        conformal_factor, inv_spatial_metric, inv_extrinsic_curvature,
-        trace_extrinsic_curvature);
-
-    const auto volume_integrand = Xcts::adm_linear_momentum_volume_integrand(
-        surface_integrand, conformal_factor, deriv_conformal_factor,
-        conformal_metric, inv_conformal_metric,
-        conformal_christoffel_second_kind, conformal_christoffel_contracted);
+    // Evaluate linear momentum volume integral
+    const auto linear_momentum_surface_integrand =
+        Xcts::adm_linear_momentum_surface_integrand(
+            conformal_factor, inv_spatial_metric, inv_extrinsic_curvature,
+            trace_extrinsic_curvature);
+    const auto linear_momentum_volume_integrand =
+        Xcts::adm_linear_momentum_volume_integrand(
+            linear_momentum_surface_integrand, conformal_factor,
+            deriv_conformal_factor, conformal_metric, inv_conformal_metric,
+            conformal_christoffel_second_kind,
+            conformal_christoffel_contracted);
     for (int I = 0; I < 3; I++) {
-      total_integral.get(I) +=
-          definite_integral(volume_integrand.get(I) * get(det_jacobian), mesh);
+      linear_momentum.get(I) += definite_integral(
+          linear_momentum_volume_integrand.get(I) * get(det_jacobian), mesh);
     }
 
+    // Evaluate angular momentum volume integral
+    const auto angular_momentum_surface_integrand =
+        Xcts::adm_angular_momentum_z_surface_integrand(
+            linear_momentum_surface_integrand, inertial_coords);
+    const auto angular_momentum_volume_integrand =
+        Xcts::adm_angular_momentum_z_volume_integrand(
+            linear_momentum_volume_integrand, inertial_coords);
+    angular_momentum_z.get() += definite_integral(
+        get(angular_momentum_volume_integrand) * get(det_jacobian), mesh);
+
     // Loop over external boundaries
     for (auto boundary_direction : current_element.external_boundaries()) {
-      // Skip interfaces not at the outer boundary
+      // Skip interfaces not at the inner boundary
       if (boundary_direction != Direction<3>::lower_zeta()) {
         continue;
       }
 
       // Get interface coordinates.
-      const auto face_logical_coords =
+      auto face_logical_coords =
           interface_logical_coordinates(face_mesh, boundary_direction);
       const auto face_inv_jacobian =
           logical_to_inertial_map.inv_jacobian(face_logical_coords);
 
+      // Shift x-coordinate
+      face_logical_coords.get(0) -= x_shift;
```

Comment:
Logical coordinates are [-1, 1] on the reference cube, you shouldn't shift them. The shift is part of the map from logical to inertial coords. I think you can just delete this line, because you don't need the face coordinates below.

Reply:
Indeed, this was a typo and shouldn't be here. Thanks for catching that!